File: app/src/main/java/com/example/gooner10/nepaliechord/mainsong/MainSongContract.java
package com.example.gooner10.nepaliechord.mainsong;

import com.example.gooner10.nepaliechord.model.Song;

import java.util.List;

/**
 * Contract for the mainActivity
 */

public interface MainSongContract {
    interface MainSongView {
        void displaySong(List<Song> songList);
    }

    interface MainSongPresenter {
        void fetchSong();
    }
}

File: app/src/main/java/com/example/gooner10/nepaliechord/mainsong/MainSongContract.kt
package com.example.gooner10.nepaliechord.mainsong

import com.example.gooner10.nepaliechord.model.Song

/**
 * Contract for the mainActivity
 */

interface MainSongContract {
    interface MainSongView {
        fun displaySong(songList: MutableList<Song>)
    }

    interface MainSongPresenter {
        fun fetchSong()

        fun onDataFetched(songList: MutableList<Song>)
    }
}
